Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,135 @@
+                        logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+                else
+                {
+                    logger.LogDebug("Environment variable value does not start with '['.");
+                    var config = JsonSerializer.Deserialize<ApiGatewayRouteConfig>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.Add(config);
+                        logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error deserializing environment variable {key}: {ex.Message}");
+                logger.LogDebug("Error deserializing environment variable {Key}: {Message}", key, ex.Message);
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public ApiGatewayRouteConfig? GetRouteConfig(string httpMethod, string path)
+    {
+        foreach (var routeConfig in _routeConfigs)
+        {
+            _logger.LogDebug("Checking if '{Path}' matches '{Template}'.", path, routeConfig.Path);
+            var template = TemplateParser.Parse(routeConfig.Path);
+
+            var matcher = new TemplateMatcher(template, GetDefaults(template));
+
+            var routeValueDictionary = new RouteValueDictionary();
+            if (!matcher.TryMatch(path, routeValueDictionary))
+            {
+                _logger.LogDebug("'{Path}' does not match '{Template}'.", path, routeConfig.Path);
+                continue;
+            }
+            
+            _logger.LogDebug("'{Path}' matches '{Template}'. Now checking the HTTP Method.", path, routeConfig.Path);
+            
+            if (!routeConfig.HttpMethod.Equals(httpMethod, StringComparison.InvariantCultureIgnoreCase))
+            {
+                _logger.LogDebug("HTTP Method of '{Path}' is {HttpMethod} and does not match the method of '{Template}' which is {TemplateMethod}.", path, httpMethod, routeConfig.Path, routeConfig.HttpMethod);
+                continue;
+            }
+            
+            _logger.LogDebug("{HttpMethod} {Path} matches the existing configuration {TemplateMethod} {Template}.", httpMethod, path, routeConfig.HttpMethod, routeConfig.Path);
+
+            return routeConfig;
+        }
+
+        return null;
+    }
+
+    private RouteValueDictionary GetDefaults(RouteTemplate parsedTemplate)
```

Comment:
Are defaults even a think in API Gateway routing? I thought with API Gateway routing you can just have the name of the variable. I could be wrong but be sure we are not just implementing a feature of the Microsoft library and not API Gateway.

Reply:
You are right. The function was not doing anything for our use case. I removed it.